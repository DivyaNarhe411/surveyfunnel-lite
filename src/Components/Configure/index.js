import React, { useEffect, useState } from 'react'
import fetchData from '../../HelperComponents/fetchData';

export default function Configure() {
	const [metaInfo, setMetaInfo] = useState({
		title: '',
		description: ''
	});

	const [companyBranding, setCompanyBranding] = useState(true);

	useEffect(() => {
		const ajaxSecurity = document.getElementById('ajaxSecurity').value;
        const post_id = new URLSearchParams(window.location.search).get('post_id');
        const data = {
            security: ajaxSecurity,
            action: 'wpsf_get_configuration_data',
            post_id,
        };
        const ajaxURL = document.getElementById('ajaxURL').value;
        fetchData( ajaxURL, data )
        .then(data => {
            if ( data.data.configure === '' ) {
				return;
			}
			let configure = JSON.parse(data.data.configure);
			setMetaInfo(configure.metaInfo);
			setCompanyBranding(configure.companyBranding);
        })
	}, []);

	const handleMetaChange = (e) => {
		setMetaInfo({
			...metaInfo,
			[e.target.name]: e.target.value,
		});
	}

	const saveConfiguration = () => {
		const ajaxSecurity = document.getElementById('ajaxSecurity').value;
        const post_id = new URLSearchParams(window.location.search).get('post_id');
        const data = {
            security: ajaxSecurity,
            action: 'wpsf_save_configuration_data',
            post_id,
			configuration: JSON.stringify({metaInfo, companyBranding})
        };
        const ajaxURL = document.getElementById('ajaxURL').value;
        fetchData( ajaxURL, data )
        .then(data => {
            console.log(data);
        })
	}

	return (
		<div className="Configure">
			<div className="configurationLeft">
				General Settings
			</div>
			<div className="configurationRight">
				<div className="meta-info">
					<h3>Meta Information</h3>
					<p>Used by social networks and search engines</p>
					<label htmlFor="title">Title: </label>
					<input type="text" value={metaInfo.title} name="title" onChange={handleMetaChange} />
					<label htmlFor="description">Description: </label>
					<textarea value={metaInfo.description} name="description" cols="30" rows="10" onChange={handleMetaChange}></textarea>
					<h3>Company Branding</h3>
					<p>Logo visible on deployed content: </p>
					<input type="checkbox" checked={companyBranding} onChange={() => {setCompanyBranding(!companyBranding)}} />
					<button type="button" onClick={saveConfiguration}>save</button>
				</div>
			</div>
		</div>
	)
}
