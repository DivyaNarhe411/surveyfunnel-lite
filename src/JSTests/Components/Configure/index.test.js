/**
 * @jest-environment jsdom
 */
import React from 'react';
import renderer from 'react-test-renderer';
import Configure from '../../../Components/Configure';

test('Configure Component test', () => {

	document.body.innerHTML = `
		<input type="hidden" id="ajaxURL" value="ajaxUrl" />
		<input type="hidden" id="ajaxSecurity" value="ajaxSecurity" />
	`;

	const component = renderer.create( <Configure /> );
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});