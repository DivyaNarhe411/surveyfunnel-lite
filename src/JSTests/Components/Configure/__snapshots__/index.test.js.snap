// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Configure Component test 1`] = `
<div
  className="configure"
>
  <div
    className="configurationLeft"
  >
    <div
      className="configurationLeft-element"
    >
      <div
        className="configurationLeft-element-title"
      >
        <h3>
          General
        </h3>
        <p>
          Basic settings
        </p>
      </div>
      <img
        src="test-file-stub"
      />
    </div>
  </div>
  <div
    className="configurationRight"
  >
    <div
      className="meta-info"
    >
      <div
        className="configure-fields"
      >
        <div
          className="meta-info-container"
        >
          <h3>
            Meta Information
          </h3>
          <p>
            Used by social networks and search engines
          </p>
          <label
            htmlFor="title"
          >
            Title: 
          </label>
          <input
            name="title"
            onChange={[Function]}
            type="text"
            value=""
          />
          <label
            htmlFor="description"
          >
            Description: 
          </label>
          <textarea
            cols="30"
            name="description"
            onChange={[Function]}
            rows="10"
            value=""
          />
        </div>
        <div
          className="company-name-container"
        >
          <h3>
            Company Branding
          </h3>
          <div
            className="companyLogoBox"
          >
            <p>
              Logo visible on deployed content:
            </p>
            <input
              checked={true}
              id="useCompanyLogo"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              htmlFor="useCompanyLogo"
            >
               
            </label>
          </div>
        </div>
      </div>
      <div
        className="meta-info-save-container"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          Save
        </button>
      </div>
    </div>
  </div>
</div>
`;
